### PR TITLE
REG-2544 `executePromiseOrTimeout` is suddenly needed

### PR DESCRIPTION
**Pull request by cantremember**
- implemented per @madhur's https://github.com/joylifeinc/event_service/pull/498
- because it's needed by the Health Check aggregator, which must self-timeout
- also, fixed a `nock` lifecycle issue

while i was here, since our logging was weak;

- proxy the Session ID off to `identity_servide` during #me()
- create a traceable name for the Query

### CODE REVIEW

File: src/server/apollo.context.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ const USER = gql`
 
 const IDENTITY_ME = gql`
   ${USER}
-  query GetMe {
+  query GetMeApolloContext {
```
Comment (cantremember) [inline src/server/apollo.context.ts:56]: the more time i spend in Apollo Studio, the more i want to see these names be representative

File: src/server/apollo.context.ts [hunk 2]
```diff
@@ -392,7 +392,11 @@ export class Context
     const telemetryHeaders = deriveTelemetryHeadersFromContext(telemetry.context());
 
     const token = (overrides && overrides.token) || this.token;
-    const headers = Object.assign(telemetryHeaders, (overrides && overrides.headers));
+    const headers = omitBy({
+      ...telemetryHeaders,
+      [ SESSION_HEADER_NAME ]: this.sessionId,
+      ...overrides?.headers,
+    }, isNil);
```
Comment (cantremember) [inline src/server/apollo.context.ts:399]: it was difficult to trace `Context#me` calls back to the orignating Gateway request -- they appeared as if they came from multiple Session IDs

File: src/utils/healthCheck/requestHandler.ts [hunk 3]
```diff
@@ -27,7 +29,12 @@ export function createHealthCheckRequestHandler<T extends Context = Context>(
     await Promise.all(Object.keys(checkers).map(async (key) => {
       const checker = checkers[key];
       try {
-        resultsByKey[key] = await checker(context);
+        // don't wait forever;
+        //   if we eventually get a success, we would appear to be healthy to our logs,
+        //   however ... k8s will have given up waiting, and we're actually failed
+        //   and been taken out of the load pool, or out behind the barn.
+        //   we need visiblity into our failure
+        resultsByKey[key] = await executePromiseOrTimeout(timeoutMs, checker(context));
```
Comment (cantremember) [inline src/utils/healthCheck/requestHandler.ts:37]: this turns out to be hella important

**four** of our Kubenetes Nodes were unable to reach the Production Elasticsearch -- for some unknown reason.  the Nodes have been cycled.  in the meanwhile,
- `product_service` Pods in those Nodes would fall out of the load pool
- leading to other intermittent failures in `stitch_service`
- which became a torrent of blood and froth which affected all Joy Catalog experiences

File: test/helpers/globalLifecycleNock.ts [hunk 4]
```diff
@@ -32,13 +32,17 @@ afterEach(() => {
...
+  try {
+    // https://github.com/nock/nock#isdone
+    //   enforce that all mocks have been called
+    //   (nope, you can't get access to the `nock.Scope` itself -- you just get its 'name')
+    assert.ok(nock.isDone(), `nock.pendingMocks: ${ JSON.stringify(nock.pendingMocks()) }`);
+  }
+  finally {
+    // https://github.com/nock/nock#enabling-requests
+    nock.cleanAll();
+    nock.enableNetConnect();
```
Comment (cantremember) [inline test/helpers/globalLifecycleNock.ts:44]: yeah.  i'd fixed this in a lot of other places already

**Review by cantremember — COMMENTED** [review]
## annotated

File: src/utils/healthCheck/requestHandler.ts (hunk 3 above)
Comment (mcorrigan89) [reply src/utils/healthCheck/requestHandler.ts:37]: Torrent of blood and froth, this is a bad thing, correct?

**Review by mcorrigan89 — APPROVED** [review]

**Pull request merged**